### PR TITLE
Configure CI

### PR DESCRIPTION
**Pull request by Shatur**
I set up dependabot (to automatically update dependencies), [cargo deny](https://github.com/EmbarkStudios/cargo-deny) (to check dependencies for vulnerabilities and license compatibility), formatting, tests and clippy checks.
Also I fixed discovered warnings (all changes was done only to silence warnings), applied formatting to make CI pass.

**Note:** I had to upgrade eframe to avoid security issue in nix crate (detected by cargo deny). Also in new version light or dark mode detected from the system, so I fixed colors a little to make it look nice in both modes.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+          args: --no-deps --all-features -- -D warnings
+
+  format:
+    name: Format
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone repo
+        uses: actions/checkout@v3
+
+      - name: Format
```
Comment (lucaspoffo) [inline .github/workflows/main.yml:67]: Do we not need to add `rustfmt` to run this step? Like: 

```suggestion
      - uses: actions-rs/toolchain@v1
        with:
          profile: minimal
          toolchain: stable
          override: true
          components: rustfmt

      - name: Format
```

Comment (Shatur) [reply .github/workflows/main.yml:67]: Not sure if I get it. We have it down below: https://github.com/gardum-game/renet/blob/1c073afa29f7cd8c4af453fe831d1428d0173210/.github/workflows/main.yml#L70

You can see it in action: https://github.com/gardum-game/renet/actions/runs/2144003508

**Comment by lucaspoffo** [comment]
Thanks! I'm not used to github actions, so this helps a bunch.

**Comment by Shatur** [comment]
You are welcome!

**Pull request merged**